joplin/js/CreateContentModal/index.js

```diff
@@ -77,13 +85,28 @@ class CreateContentModal extends Component {
     this.setState({ topic: id });
   };
 
-  redirectToEditPage = () => {
-    this.writeToLocalStorage();
-    window.location.href = this.state.redirectUrl;
+  redirectToEditPage = id => {
+    window.location.href = `/admin/pages/${id}/edit/`;
   };
 
-  writeToLocalStorage = () => {
-    localStorage.wagtailCreateModal = JSON.stringify(this.state);
+  createPage = () => {
+    axios
+      .post(
+        '/admin/pages/new_from_modal/',
+        {
+          type: this.state.type,
+          title: this.state.title,
+          topic: this.state.topic,
+        },
+        { headers: { 'X-CSRFToken': Cookies.get('csrftoken') } },
+      )
+      .then(response => {
+        this.redirectToEditPage(response.data.id);
+      })
+      .catch(error => {
+        console.log(error);
```

Comment:
Same here with error states. We probably should alert the user when the create page fails and provide them an alternative?